File: Sequence Valley/String/Q1. Detect Capital/Q1_DetectCapital.java
class Solution {                                       // Define the Solution class
    public boolean detectCapitalUse(String word) {     // Method to check capital usage
        
        if (word.length() == 0 || word.length() == 1)  // If word is empty or has one letter
            return true;                               // Always valid
        
        if (Character.isUpperCase(word.charAt(0))) {  // If first letter is uppercase
            
            boolean isFirstCharacter =                // Check case of second letter
                    Character.isUpperCase(word.charAt(1));
            
            for (int i = 2; i < word.length(); i++) { // Loop from third character
                boolean currentCharState =            // Check current letter case
                        Character.isUpperCase(word.charAt(i));
                
                if (currentCharState != isFirstCharacter) // If case does not match
                    return false;                     // Capital usage is wrong
            }
        } else {                                      // If first letter is lowercase
            
            for (int i = 1; i < word.length(); i++) { // Loop through remaining letters
                if (Character.isUpperCase(word.charAt(i))) // If any letter is uppercase
                    return false;                     // Capital usage is wrong
            }
        }
        
        return true;                                  // Capital usage is correct
    }
}

File: Sequence Valley/String/Q1. Detect Capital/Q1_DetectCapital.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        if len(word) < 2:                      # If word has 0 or 1 letter
            return True                        # Always valid

        if word[0].isupper() and word[1].isupper():  # If first two letters are uppercase
            for i in range(2, len(word)):      # Check remaining letters
                if word[i].islower():           # If any letter is lowercase
                    return False                # Invalid capital usage
        else:
            for i in range(1, len(word)):      # Check from second letter
                if word[i].isupper():           # If any letter is uppercase
                    return False                # Invalid capital usage

        return True                             # Capital usage is correct
